### PR TITLE
Remove hash.js and elliptic dependenices for passport-azure-ad

### PR DESCRIPTION
**Pull request by pkanher617**
Replaces the `jwk-to-pem` library with the `node-jose` library

### CODE REVIEW

**Review by peterzenz — APPROVED** [review]

**Review by tnorling — APPROVED** [review]

**Review by jasonnutter — APPROVED** [review]

File: maintenance/passport-azure-ad/lib/jwe.js [hunk 1]
```diff
@@ -507,17 +507,34 @@ exports.decrypt = (jweString, jweKeyStore, log, callback) => {
      *  cek decryption
      ***************************************************************************/
     var cek_result = decryptCEK(header, encrypted_cek, jweKeyStore, log);
-    if (cek_result.error)
+    if (cek_result.error) {
+      if (cek_result.error === "converting jwk to RSA privatePemKey to decrypt cek") {
+        jose.JWK.asKey(cek_result.cek, "pem").then((pemKey) => {
```
Comment (jasonnutter) [inline maintenance/passport-azure-ad/lib/jwe.js:512]: I'm not sure what the expected support for es6 syntax is, but might want to have this be a non-arrow function just to be safe.

File: maintenance/passport-azure-ad/test/Chai-passport_test/jwe_test.js [hunk 2]
```diff
@@ -18,11 +18,10 @@ describe('positive test', function() {
...
 
-    var decrypted;
-    jwe.decrypt(jweString, jwk, log, (error, decrypted_token) => { decrypted = decrypted_token; });
-
-    expect(decrypted).to.equal('You can trust us to stick with you through thick and thin–to the bitter end. And you can trust us to keep any secret of yours–closer than you keep it yourself. But you cannot trust us to let you face trouble alone, and go off without a word. We are your friends, Frodo.');
-    done();
+    jwe.decrypt(jweString, jwk, log, (error, decrypted_token) => {
+      expect(decrypted_token).to.equal('You can trust us to stick with you through thick and thin–to the bitter end. And you can trust us to keep any secret of yours–closer than you keep it yourself. But you cannot trust us to let you face trouble alone, and go off without a word. We are your friends, Frodo.');
```
Comment (sameerag) [inline maintenance/passport-azure-ad/test/Chai-passport_test/jwe_test.js:22]: love this :-)

**Review by sameerag — APPROVED** [review]

File: maintenance/passport-azure-ad/lib/jwe.js (hunk 1 above)
Comment (pkanher617) [reply maintenance/passport-azure-ad/lib/jwe.js:512]: We use arrow functions in multiple places in passport today, is there a specific concern with the arrow function here?

Comment (jasonnutter) [reply maintenance/passport-azure-ad/lib/jwe.js:512]: Nope, just wanted to be sure given this lib doesnt use Typescript or any other transpiling.

**Pull request merged**